addons/account/models/account_payment.py

```diff
@@ -968,10 +968,16 @@ def unmark_as_sent(self):
 
     def action_post(self):
         ''' draft -> posted '''
-        # Do not allow to post if the account is required but not trusted
-        for payment in self:
-            if payment.require_partner_bank_account and not payment.partner_bank_id.allow_out_payment:
-                raise UserError(_('To record payments with %s, the recipient bank account must be manually validated. You should go on the partner bank account in order to validate it.', self.payment_method_line_id.name))
```

Comment:
when fixing stable versions, any change in translated strings will break existing translations. So that sucks, but your solution looks worse than the current one. 

Something like below might do
```diff
-                raise UserError(_('To record payments with %s, the recipient bank account must be manually validated. You should go on the partner bank account in order to validate it.', self.payment_method_line_id.name))
+                raise UserError(_('To record payments with %s, the recipient bank account must be manually validated. You should go on the partner bank account in order to validate it.', payment.partner_id.name))
```

Reply:
The traceback is in the linked issue, I was not sure if we can change the texts or not, I'll do this instead.